### PR TITLE
fixes #17407 - add a new hosts dashboard widget

### PR DESCRIPTION
**Pull request by timogoebel**
This commit adds a small dashboard that shows a list of the latest eight hosts that were created. Nothing too fancy.

### CODE REVIEW

**Comment by timogoebel** [comment]
Tests failures look unrelated.

File: app/views/dashboard/_new_hosts_widget.html.erb [hunk 1]
```diff
@@ -0,0 +1,11 @@
+<% hosts = Host.authorized(:view_hosts).includes(:owner, :operatingsystem).order(:created_at => :desc).limit(8) %>
```
Comment (ohadlevy) [inline app/views/dashboard/_new_hosts_widget.html.erb]: does it also make sense to do something based on installed_at attribute as well?

File: app/views/dashboard/_new_hosts_widget_host_list.html.erb [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+  <thead>
+    <th><%= _('Host') %></th>
+    <th><%= _('Operating system') %></th>
+    <th><%= _('Owner') %></th>
+    <th><%= _('Created') %></th>
+  </thead>
+  <tbody>
+    <% hosts.each do |host| %>
+      <tr>
+        <td><%= link_to trunc_with_tooltip(h(host.name)), host_path(host) %></td>
```
Comment (ohadlevy) [inline app/views/dashboard/_new_hosts_widget_host_list.html.erb]: nitpick: h is not longer needed.

File: app/views/dashboard/_new_hosts_widget.html.erb (hunk 1 above)
Comment (tbrisker) [reply app/views/dashboard/_new_hosts_widget.html.erb]: the dashboard data service already loads hosts respecting authorization and filters, you can use `@data.hosts.includes...`  instead and get the widget to respect the filter.

File: app/views/dashboard/_new_hosts_widget_host_list.html.erb [hunk 3]
```diff
@@ -0,0 +1,18 @@
+<table class="table table-striped ellipsis">
```
Comment (tbrisker) [inline app/views/dashboard/_new_hosts_widget_host_list.html.erb]: `ellipsis` class should be on the `<td>`, not the table. the table should have `table-fixed` and the `th` elements should set their width.

File: app/views/dashboard/_new_hosts_widget_host_list.html.erb (hunk 2 above)
Comment (tbrisker) [reply app/views/dashboard/_new_hosts_widget_host_list.html.erb]: `trunc_with_tooltip` is not needed if using `ellipsis` on the `td`.

**Comment by timogoebel** [comment]
@ohadlevy, @tbrisker : Thanks for your comments. I added a "installed_at" column and fixed the ellipsis code.

**Comment by dLobatog** [comment]
@timogoebel 👍 by me, however I'll wait until @tbrisker takes a final look since he's been the one reviewing this. Thank you 😄

File: app/views/dashboard/_new_hosts_widget.html.erb [hunk 4]
```diff
@@ -0,0 +1,11 @@
+<% hosts = @data.hosts.includes(:operatingsystem).preload(:owner).order(:created_at => :desc).limit(8) %>
```
Comment (tbrisker) [inline app/views/dashboard/_new_hosts_widget.html.erb]: any reason not to use `.joins(:operatingsystem, :owner)` and getting the data in one call? also seems like there is room for 9 rows in the widget.

File: app/views/dashboard/_new_hosts_widget_host_list.html.erb [hunk 5]
```diff
@@ -0,0 +1,20 @@
...
+    <th width="30%"><%= _('Host') %></th>
+    <th><%= _('Operating system') %></th>
+    <th><%= _('Owner') %></th>
+    <th><%= _('Created') %></th>
+    <th><%= _('Installed') %></th>
+  </thead>
+  <tbody>
+    <% hosts.each do |host| %>
+      <tr>
+        <td class="ellipsis"><%= link_to host.name, host_path(host) %></td>
```
Comment (tbrisker) [inline app/views/dashboard/_new_hosts_widget_host_list.html.erb:12]: Ellipsis on the dashboard seems broken but it is unrelated to this. Opened http://projects.theforeman.org/issues/17501 to handle that.

**Review by tbrisker — CHANGES_REQUESTED** [review]
Mostly looks good, just a couple of inline comments.

File: app/views/dashboard/_new_hosts_widget.html.erb (hunk 4 above)
Comment (timogoebel) [reply app/views/dashboard/_new_hosts_widget.html.erb]: Yes, rails joins seem to be broken for polymorphic associations. You can try that yourself, just use `includes(:operatingsystem, :owner)` and add a filter that doesn't match any host.

File: app/views/dashboard/_new_hosts_widget_host_list.html.erb (hunk 5 above)
Comment (timogoebel) [reply app/views/dashboard/_new_hosts_widget_host_list.html.erb:12]: Thanks.

File: app/views/dashboard/_new_hosts_widget.html.erb (hunk 4 above)
Comment (timogoebel) [reply app/views/dashboard/_new_hosts_widget.html.erb]: For the record, rails throws a `ActiveRecord::EagerLoadPolymorphicError: Cannot eagerly load the polymorphic association :owner` error.

**Comment by timogoebel** [comment]
@tbrisker : Added one more row to the dashboard.

**Review by tbrisker — APPROVED** [review]

**Comment by tbrisker** [comment]
good to merge once :green_apple:

**Pull request merged**